synapse/appservice/scheduler.py

```diff
@@ -114,7 +122,19 @@ def enqueue(self, service, event):
             "as-sender-%s" % (service.id,), self._send_request, service
         )
 
-    async def _send_request(self, service):
+    def enqueue_ephemeral(self, service: ApplicationService, events: List[Any]):
+        self.queued_ephemeral.setdefault(service.id, []).extend(events)
+
+        # start a sender for this appservice if we don't already have one
+
+        if service.id in self.requests_in_flight:
+            return
+
+        run_as_background_process(
+            "as-sender-%s" % (service.id,), self._send_request, service
```

Comment:
Should this differ from the name used in `enqueue`?

Reply:
I'm not sure it needs to? I've instead chosen to remove the duplicate code and make it one function.